server/s3_put_chunk_upload_object_action.cc

```diff
@@ -208,11 +208,13 @@ void S3PutChunkUploadObjectAction::parse_x_amz_tagging_header(
     for (struct evkeyval* header = key_value.tqh_first; header;
          header = header->next.tqe_next) {
 
-      decoded_key = evhttp_decode_uri(header->key);
+      decoded_key = evhttp_uridecode(header->key, 0, NULL);
```

Comment:
does this set decoded_key to null if evhttp_uriecode fails

Reply:
Yes.